tidb-cloud/v7.1.0-performance-benchmarking-with-sysbench.md

```diff
@@ -0,0 +1,175 @@
+
+The TiKV parameter [`prefill-for-recycle`](https://docs.pingcap.com/tidb/stable/tikv-configuration-file#prefill-for-recycle-new-in-v700) can make log recycling effective immediately after initialization.
+
+Enable the [`prefill-for-recycle`](https://docs.pingcap.com/tidb/stable/tikv-configuration-file#prefill-for-recycle-new-in-v700) parameter in `oltp_insert`, `oltp_read_write` , `oltp_update_index`, and `oltp_update_non_index` tests:
+
+```yaml
+raft-engine.prefill-for-recycle = true
+```
+
+Disable the [`prefill-for-recycle`](https://docs.pingcap.com/tidb/stable/tikv-configuration-file#prefill-for-recycle-new-in-v700) parameter in the `oltp_point_select` test:
+
+```yaml
+raft-engine.prefill-for-recycle = false
+```
+
+### Benchmark executor
+
+The benchmark executor sends SQL queries to the TiDB cluster. In this test, its hardware configuration is as follows:
+
+- Machine type: Amazon EC2 (us-west-2)
+- Instance type: c6a.2xlarge
+- Sysbench version: sysbench 1.0.20 (using bundled LuaJIT 2.1.0-beta2)
+
+## Test steps
+
+This section introduces how to perform the Sysbench performance test step by step.
+
+1. In the [TiDB Cloud console](https://tidbcloud.com/), create a TiDB Dedicated cluster that meets the [test environment](#tidb-cluster) requirements.
+
+   For more information, see [Create a TiDB cluster](/tidb-cloud/create-tidb-cluster.md).
+
+2. On the benchmark executor, connect to the newly created cluster and create a database named `sbtest`.
+
+   To connect to the cluster, see [Connect via Private Endpoint](/tidb-cloud/set-up-private-endpoint-connections.md).
+
+   To create the `sbtest` database, execute the following SQL statement:
+
+   ```sql
+   CREATE DATABASE sbtest;
+   ```
+
+3. Load Sysbench data to the `sbtest` database.
+
+   1. The test in this document is implemented based on [sysbench](https://github.com/akopytov/sysbench). To install sysbench, see [Building and installing from source](https://github.com/akopytov/sysbench#building-and-installing-from-source).
+
+   2. Run the following `sysbench prepare` command to import 32 tables and 10,000,000 rows to the `sbtest` database. Replace `${HOST}`, `${PORT}`, `${THREAD}`, and `${PASSWORD}` with your actual values.
+
+      ```shell
+      sysbench oltp_common \
+         --threads=${THREAD} \
+         --db-driver=mysql \
+         --mysql-db=sbtest \
+         --mysql-host=${HOST} \
+         --mysql-port=${PORT} \
+         --mysql-user=root \
+         --mysql-password=${PASSWORD} \
+         prepare --tables=32 --table-size=10000000
+      ```
+
+4. Run the following `sysbench run` command to conduct stress tests on the TiDB Dedicated cluster. For each concurrency, the test takes 20 minutes. This document conducts tests on five workloads: `oltp_point_select`, `oltp_read_write`, `oltp_update_non_index`, `oltp_update_index`, and `oltp_insert`. For each workload, this document conducts three tests with the `${THREAD}` value of `100`, `200`, and `400`.
```

Comment:
"For each concurrency", what does "each concurrency" mean?

Reply:
> `--time` limits the total execution time in seconds.

For each workload, the following `sysbench run` command is executed 3 times, each time with a different level of concurrency (the number of threads) specified by `${THREAD}`.
```
sysbench ${WORKLOAD} run \
      --mysql-host=${HOST} \
      --mysql-port=${PORT} \
      --mysql-user=root \
      --db-driver=mysql \
      --mysql-db=sbtest \
      --threads=${THREAD} \
      --time=1200 \
      --report-interval=10 \
      --tables=32 \
      --table-size=10000000 \
      --mysql-ignore-errors=1062,2013,8028,9007 \
      --auto-inc=false \
      --mysql-password=${PASSWORD}
```